controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/AbstractRequestProcessor.java

```diff
@@ -124,10 +124,15 @@ public String getProcessorName() {
         CompletableFuture<Void> resultFuture = new CompletableFuture<>();
 
         OperationContext context = streamMetadataStore.createContext(scope, stream);
-        suppressException(streamMetadataStore.getWaitingRequestProcessor(scope, stream, context, executor), null,
-                "Exception while trying to fetch waiting request. Logged and ignored.")
-                .thenAccept(waitingRequestProcessor -> {
-                    if (waitingRequestProcessor == null || waitingRequestProcessor.equals(getProcessorName())) {
+        CompletableFuture<String> waitingProcFuture = suppressException(streamMetadataStore.getWaitingRequestProcessor(scope, stream, context, executor), null,
+                "Exception while trying to fetch waiting request. Logged and ignored.");
+        CompletableFuture<Boolean> ignoreFairnessFuture = task.ignoreFairness(event);
+        
+        CompletableFuture.allOf(waitingProcFuture, ignoreFairnessFuture)
+                .thenAccept(v -> {
+                    boolean ignoreFairness = ignoreFairnessFuture.join();
+                    String waitingRequestProcessor = waitingProcFuture.join();
+                    if (ignoreFairness || waitingRequestProcessor == null || waitingRequestProcessor.equals(getProcessorName())) {
```

Comment:
`ignoreFairness` sounds bad, what we really need a predicate like `shouldProceed`, which evaluates to `true` in the case that:

- there is no other blocked processor
- there is a blocked processor, which won't be unblocked if we block the present one

and evaluates to `false` otherwise.

Reply:
I think a better name would be - `isAlreadyStarted` as that is most explicitly stating what we are doing here.

Reply:
I'm fine with the suggestion.